packages/light-apps/src/Alerts/Alerts.tsx

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018-2019 @paritytech/substrate-light-ui authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { AppContext } from '@substrate/ui-common';
+import { Alert } from '@substrate/ui-components';
+import React from 'react';
+
+export class Alerts extends React.PureComponent {
+  static contextType = AppContext;
+
+  context!: React.ContextType<typeof AppContext>; // http://bit.ly/typescript-and-react-context
+
+  handleDismiss = () => {
+    const { alertStore } = this.context;
+
+    // FIXME We suppose that there's only one Alert at a time for now. If/when
+    // we have multiple alert, we should call alertStore.removeNth().
+    alertStore.dequeue();
+  }
+
+  render () {
+    const { alertStore } = this.context;
+
+    if (!alertStore.alerts) {
+      return null;
+    }
+
+    return (
+      <React.Fragment>
+        {alertStore.alerts.map((alert, index) => (
+          <Alert
+            error={alert.type === 'error'}
+            info={alert.type === 'info'}
+            key={index} // FIXME Use a real ID
```

Comment:
I think if we're dequeuing an alert and enqueuing another one right after, we'll still have one Alert (although different) rendered with the same key 0, so the new alert probably wouldn't get displayed. I guess a way to make the `key` unique is to store the amount of mutations in alertStore (how many times we dequeued or enqueued), and as a key to the `<Alert>` use ``{`${alertStore.mutationCount}-${index}`}`` but that's not very pretty

Reply:
Hmm, I agree it's not pretty. Some other ideas:

- Let the developer add an `id` field here: https://github.com/paritytech/substrate-light-ui/issues/246. He makes sure it's unique.
- Add an internal counter in AlertStore, on each new alert ++counter

Reply:
internal counter idea sounds good to me!

> Let the developer add an id field here: https://github.com/paritytech/substrate-light-ui/pull/253/files#diff-a60e3ebec2c8d7ba9e576b16baa892baR76. He makes sure it's unique.

sounds very tedious: the caller would have to generate a different ID each time, otherwise there could not be the same alert twice